src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -0,0 +1,233 @@
+"""
+FEWS NET processing.
+
+Download and save the data provided by FEWS NET.
+"""
+
+import logging
+import zipfile
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Union
+
+from requests.exceptions import HTTPError
+
+from aatoolbox.utils.io import download_url, unzip
+
+logger = logging.getLogger(__name__)
+BASE_URL_COUNTRY = (
+    "https://fdw.fews.net/api/ipcpackage/"
+    "?country_code={iso2}&collection_date={YYYY}-{MM}-01"
+)
+BASE_URL_REGION = (
+    "https://fews.net/data_portal_download/download"
+    "?data_file_path=http://shapefiles.fews.net.s3.amazonaws.com/"
+    "HFIC/{region_code}/{region_name}{YYYY}{MM}.zip"
+)
+
+
+def download_zip(
+    url: str,
+    zip_filename: str,
+    output_dir: Path,
+) -> bool:
+    """
+    Download and unzip the file at the url.
+
+    Parameters
+    ----------
+    url : str
+        url that contains the zip file to be downloaded
+    zip_filename : str
+        name of the zipfile
+    output_dir : Path
+        path of dir to which the zip content should be written
+
+    Returns
+    -------
+    valid_file: bool
+        if True, the url contains a valid zip file
+
+    """
+    valid_file = False
+
+    try:
+        # create tempdir to write zipfile to
+        with TemporaryDirectory() as temp_dir:
+            zip_path = Path(temp_dir) / zip_filename
+            download_url(url=url, save_path=zip_path)
+            logger.info(f"Downloaded {url} to {zip_path}")
+
+            try:
+                unzip(zip_file_path=zip_path, save_dir=output_dir)
+                logger.debug(f"Unzipped to {output_dir}")
+                valid_file = True
+            except zipfile.BadZipFile:
+                # indicates that the url returned something that wasn't a
+                # zip, happens often and indicates data for the given
+                # country - date is not available
+                logger.debug(
+                    f"No zip data returned from url {url} "
+                    f"check that the area and date exist."
+                )
+
+    except HTTPError as e:
+        logger.info(e)
```

Comment:
Wow you are so good, thanks for all this help. I am having a bit hard time imaging all the cases that this might occur. But what I have encountered so far is when the url is incorrect and thus gives a 404. In that case it should indeed exit and then it makes sense to not catch the error so changed that